### ISSUE TITLE
strict mode errors for nested properties

### ISSUE THREAD

**Issue by UnquietCode** [handlebars-lang/handlebars.js#1382]
I have a simple failing case for using strict mode and accessing a nested property. This is probably related to #1138 

If you look at this:
https://github.com/wycats/handlebars.js/blob/680ec96/lib/handlebars/runtime.js#L71

The code will only error if property isn't nested. If the property is nested, you will instead receive
`Cannot use 'in' operator to search for 'xyz' in undefined`

Strict mode should work on nested properties, otherwise I'm unsure of its utility.

**Comment by cwingrav** [comment]
I'm running into this too. A simple fix would be to edit the strict function, adding this line right before the current `if ( !(name in obj)) {` line:

```
if ( obj == null ) { throw new _exception2['default']('"' + name + '" not defined in null object'); }
```

**Comment by jaylinski** [comment]
It seems this was already fixed in #1527.
